### PR TITLE
Rename `Version` message to  `Handshake`

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

https://github.com/ava-labs/avalanchego/pull/2476#discussion_r1424591405

This message contains a lot more than just the version now.

## How this works

Rename.

## How this was tested

Existing UT.

### CODE REVIEW

**Review by joshua-kim — APPROVED** [review]

**Review by dhrubabasu — APPROVED** [review]

File: proto/p2p/p2p.proto [hunk 1]
```diff
@@ -91,17 +91,17 @@ message Pong {
...
+// Handshake is the first outbound message sent to a peer when a connection is
 // established to start the p2p handshake.
 //
-// Peers must respond to a Version message with a PeerList message to allow the
+// Peers must respond to a Handshake message with a PeerList message to allow the
 // peer to connect to other peers in the network.
 //
 // Peers should drop connections to peers with incompatible versions.
-message Version {
+message Handshake {
```
Comment (StephenButtolph) [inline proto/p2p/p2p.proto:101]: Probably should be a separate PR - but we should probably rename `my_version_time` in here as well.

**Review by StephenButtolph — APPROVED** [review]
